standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -922,51 +928,91 @@ private void updateReplId(Connection dbConn, ReplLastIdInfo replLastIdInfo) thro
-        params = Arrays.asList(part);
-        pst = sqlGenerator.prepareStmtWithParameters(dbConn, query, params);
-        LOG.debug("Going to execute query <" + query.replaceAll("\\?", "{}") + ">", quoteString(part));
+      List<String> questions = new ArrayList<>();
+      for(int i = 0; i < partList.size(); ++i) {
+        questions.add("?");
+      }
 
-        rs = pst.executeQuery();
-        if (rs == null || !rs.next()) {
-          throw new MetaException(" partition with name " + part + " does not exist in table " +
+      List<String> queries = new ArrayList<>();
+      StringBuilder prefix = new StringBuilder();
+      StringBuilder suffix = new StringBuilder();
+      prefix.append("select \"PART_ID\" from \"PARTITIONS\" where \"TBL_ID\" = " + tblId + " and ");
+
+      // Populate the complete query with provided prefix and suffix
+      List<Integer> counts = TxnUtils.buildQueryWithINClauseStrings(conf, queries, prefix, suffix,
+              questions, "\"PART_NAME\"", true, false);
+      int totalCount = 0;
+      for (int i = 0; i < queries.size(); i++) {
+        query = queries.get(i);
+        int partCount = counts.get(i);
+
+        LOG.debug("Going to execute query <" + query.replaceAll("\\?", "{}") + ">",
+                partList.subList(totalCount, totalCount+partCount));
+        pst = dbConn.prepareStatement(query);
+        for (int j = 0; j < partCount; j++) {
+          pst.setString(j + 1, partList.get(totalCount + j));
+        }
+        totalCount += partCount;
+        prs = pst.executeQuery();
+        if (prs == null) {
+          throw new MetaException("Partition with name " + partList + " does not exist in table " +
                   catalog + "." + db + "." + table);
         }
-        long partId = rs.getLong(1);
 
-        rs = stmt.executeQuery("select PARAM_VALUE from PARTITION_PARAMS where PARAM_KEY " +
-                " = 'repl.last.id' and PART_ID = " + partId);
-        if (rs == null || !rs.next()) {
-          stmt.executeUpdate("insert into PARTITION_PARAMS values ( " + partId +
-                  " , 'repl.last.id' , " + lastReplId + ")");
-        } else {
-          stmt.executeUpdate("update PARTITION_PARAMS set PARAM_VALUE = " + lastReplId +
-                  " where PART_ID = " + partId + " and PARAM_KEY = 'repl.last.id'");
+        while (prs.next()) {
+          long partId = prs.getLong(1);
+          rs = stmt.executeQuery("select PARAM_VALUE from PARTITION_PARAMS where PARAM_KEY " +
+                  " = 'repl.last.id' and PART_ID = " + partId);
+          if (rs == null || !rs.next()) {
+            query = "insert into PARTITION_PARAMS values ( " + partId + " , 'repl.last.id' , ? )";
+          } else {
+            query = "update PARTITION_PARAMS set PARAM_VALUE = ? " +
+                    " where PART_ID = " + partId + " and PARAM_KEY = 'repl.last.id'";
+          }
+          params = Arrays.asList(lastReplId);
```

Comment:
Repeated initialisation of params with same value. Can be done at the top only once.

Reply:
done